src/d3a/models/strategy/update_frequency.py

```diff
@@ -59,8 +59,12 @@ def delete_past_state_values(self, current_market_time_slot):
             self.energy_rate_change_per_update.pop(market_slot, None)
             self.update_counter.pop(market_slot, None)
 
+    @staticmethod
+    def get_all_markets(area):
+        return area.all_markets
```

Comment:
Nit: Is this really needed? Isn't `for market in area.all_markets:` even shorter?

Reply:
It is needed in order to be able to override this method for the settlement strategy. Have a look at the SettlementTemplateStrategyBidUpdater and SettlementTemplateStrategyOfferUpdater classes, these inherit from TemplateStrategyBid/OfferUpdater classes and override only this method, in order to apply the price increase / decrease on bids / offers of settlement markets instead of the spot markets.

Reply:
Ahhh. thanks! Now I understand.